src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplyTotal = 0;
+        int buyTotal = 0;
+
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] data = line.split(",");
+                String operation = data[0];
+                int amount = Integer.parseInt(data[1]);
+
+                if ("supply".equals(operation)) {
+                    supplyTotal += amount;
+                } else if ("buy".equals(operation)) {
+                    buyTotal += amount;
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("An error occurred while writing to the file"
+                    + fromFileName, e);
```

Comment:
The error message here is misleading. It mentions 'writing to the file' instead of 'reading from the file'. This can cause confusion during debugging. Please correct the error message to accurately reflect the operation being performed.